File: src/api/LMSEntities/DataTransferObjects/LibraryCardForDetailedDto.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace LMSEntities.DataTransferObjects
{
    public class LibraryCardForDetailedDto
    {
        [Required]
        public int Id { get; set; }

        [Required]
        public string FirstName { get; set; }

        [Required]
        public string LastName { get; set; }

        [Required]
        public string Email { get; set; }

        [Required]
        public string PhoneNumber { get; set; }

        [Required]
        public string CardNumber { get; set; }

        [Required]
        public decimal Fees { get; set; }

        [Required]
        public DateTime Created { get; set; }

        [Required]
        public AddressDto Address { get; set; }

        [Required]
        public int AddressId { get; set; }

        [Required]
        public string Gender { get; set; }

        [Required]
        public string Status { get; set; }

        [Required]
        public int Age { get; set; }

        [Required]
        public string PhotoUrl { get; set; }

        [Required]
        public IList<CheckoutForListDto> Checkouts { get; set; } = new List<CheckoutForListDto>();

    }
}

File: src/spa/src/dto/models/library-card-for-detailed-dto.ts
/* tslint:disable */
/* eslint-disable */
import { AddressDto } from './address-dto';
import { CheckoutForListDto } from './checkout-for-list-dto';
export interface LibraryCardForDetailedDto {
  address?: AddressDto;
  addressId?: number;
  age?: number;
  cardNumber?: null | string;
  checkouts?: null | Array<CheckoutForListDto>;
  created?: string;
  email?: null | string;
  fees?: number;
  firstName?: null | string;
  gender?: null | string;
  id?: number;
  lastName?: null | string;
  libraryCardPhoto?: null | string;
  phoneNumber?: null | string;
  photoUrl?: null | string;
  status?: null | string;
}
